ledger/internal/prefetcher/prefetcher.go

```diff
@@ -260,21 +264,21 @@ func (p *accountPrefetcher) prefetch(ctx context.Context) {
 		maxTxnGroupEntries = p.consensusParams.MaxTxGroupSize * 8
 	}
 
-	tasksQueue := allocPreloaderQueue(len(p.groups), maxTxnGroupEntries)
+	tasksQueue := allocPreloaderQueue(len(p.txnGroups), maxTxnGroupEntries)
```

Comment:
Commenting here, but the issue is elsewhere. I don't think maxTxnGroupEntries is calculated very well.  It's way too big.
1.  It adds `p.consensusParams.MaxAppTxnAccounts + p.consensusParams.MaxAppTxnForeignApps + p.consensusParams.MaxAppTxnForeignAssets` but it should just use `p.consensusParmams.MaxAppTotalTransactionReferences` to avoid double counting.
2. Inside of allocPreloaderQueue() 2 is added in again (presumably the same 2 that was used to indicate the sender and the app data)
I'm not sure if any of this matters. I see it's used for a capacity, so it doesn't have to be precise. But then it's also set in the `preloaderTaskQueue` structure.  This all seems like we're trying to be a bit too clever about allocating capacities, and we're getting it wrong anyway.

Reply:
Detached as a separate ticket